dapp/src/components/ProjectInfo.astro

```diff
@@ -0,0 +1,188 @@
+  <div id="modal-content" class="space-y-4 w-[600px]">
+    <h2 id="modal-project-name" class="text-3xl font-bold mb-4">Project Name</h2>
+
+    <label class="block mb-2 text-base font-medium text-black">
+      Maintainers
+      <input
+        type="text"
+        id="modal-maintainers"
+        class="block p-3 w-full text-base text-black bg-white rounded-lg border shadow-sm focus:ring-black focus:border-black"
+        placeholder="List of maintainers' addresses as G...,G..."
+        required
+      />
+    </label>
+
+    <label class="block mb-2 text-base font-medium text-black">
+      URL
+      <input
+        type="text"
+        id="modal-config-url"
+        class="block p-3 w-full text-base text-black bg-white rounded-lg border shadow-sm focus:ring-black focus:border-black"
+        placeholder="Information file URL"
+        required
+      />
+    </label>
+
+    <label class="block mb-2 text-base font-medium text-black">
+      Information file hash
+      <input
+        type="text"
+        id="modal-config-hash"
+        class="block p-3 w-full text-base text-black bg-white rounded-lg border shadow-sm focus:ring-black focus:border-black"
+        placeholder="Information file hash"
+        required
+        minlength="40"
+        maxlength="40"
+      />
+    </label>
+
+    <div id="wrap-update-button">
+      <button
+        id="update-config-button"
+        class="w-full py-5 bg-zinc-900 rounded-[14px] justify-center gap-2.5 inline-flex"
+      >
+        <span class="text-center text-white text-xl font-normal leading-7">
+          Update Config
+        </span>
+      </button>
+    </div>
+  </div>
+</Modal>
+
+<script>
+  import { loadedProjectInfo, updateConfig } from "./project";
+  import { loadedPublicKey } from "./stellar-wallets-kit";
+
+  document.addEventListener("astro:page-load", () => {
+    const topicElement = document.getElementById("project-name-topic");
+    const topicDescriptionElement = topicElement?.querySelector("p");
+    const modal = document.getElementById("update-config-modal") as HTMLDialogElement;
+
+    function updateProjectInfo() {
+      const projectInfo = loadedProjectInfo();
+      if (projectInfo) {
+        console.log("projectInfo:", projectInfo);
+        if (topicDescriptionElement) {
+          topicDescriptionElement.textContent = projectInfo.name;
+        }
+        const maintainersList = document.getElementById("project-maintainers");
+        if (maintainersList) {
+          const ul = maintainersList.querySelector("ul");
+          if (ul) {
+            ul.innerHTML = '';
+            projectInfo.maintainers.forEach(maintainer => {
+              const li = document.createElement("li");
+              li.textContent = maintainer;
+              ul.appendChild(li);
+            });
+          }
+        }
+        const githubLink = document.getElementById("github-link") as HTMLAnchorElement;
+        if (githubLink) {
```

Comment:
A general question, do we need all these `if`? Because these elements would always exist no?